### ISSUE TITLE
spirv-link generates invalid debug ids with non semantic shader debug info

### ISSUE THREAD

**Issue by danginsburg** [KhronosGroup/SPIRV-Tools#5747]
I have two valid SPIR-V files (pass spirv-val) with non-semantic shader debug info that when linked together with spirv-link generate invalid SPIR-V with the following error:

```
error: line 6093: ID '5071[%5071]' has not been defined
  %5070 = OpExtInst %void %1 DebugLine %449 %465 %465 %5071 %_arr_v2float_int_3
```

This leads to a crash later in when I run the spirv-opt aggressive DCE pass on it.  I am unable to provide the shaders publicly, but I will provide them in private to @jeremy-lunarg.  I'm filing the issue publicly so that it is tracked.

**Comment by pierremoreau** [comment]
Yes, there is an issue where the code updating the IDs does not run on debug-line instructions. I have a fix ready, but still need to add new tests to validate it.

**Comment by pierremoreau** [comment]
@jeremy-lunarg I won’t have time to finish the MR tonight, but if you want to give the fix a try locally, you simply need to change [this line](https://github.com/KhronosGroup/SPIRV-Tools/blob/363486479d4c8dfbfdf2e2dc397cd10aa15f80b0/source/link/linker.cpp#L198) to `    }, true);`.

**Comment by Ryp** [comment]
Hello, just been hit by this issue as well, the suggested fix worked out.
